server/lib/tuist_web/controllers/api/bundles_controller.ex

```diff
@@ -20,6 +22,170 @@ defmodule TuistWeb.API.BundlesController do
+                 total_pages: %Schema{type: :integer}
+               },
+               required: [:current_page, :page_size, :total_count, :total_pages]
+             }
+           },
+           required: [:data, :meta]
+         }},
+      unauthorized: {"You need to be authenticated to list bundles", "application/json", Error},
+      forbidden: {"You are not authorized to list bundles", "application/json", Error}
+    }
+
+  def index(%{assigns: %{selected_project: selected_project}} = conn, params) do
+    # Create filters in the format Flop expects (list of filter objects)
+    filters = [
+      %{field: :project_id, op: :==, value: selected_project.id}
+    ]
+
+    filters =
+      case Map.get(params, "git_branch") do
+        nil -> filters
+        branch -> [%{field: :git_branch, op: :==, value: branch} | filters]
+      end
+
+    flop_params = %{
+      filters: filters,
+      order_by: [:inserted_at],
+      order_directions: [:desc],
+      page_size: Map.get(params, "page_size", 50)
+    }
+
+    flop_params =
+      case Map.get(params, "page") do
+        nil -> flop_params
+        page -> Map.put(flop_params, :page, page)
+      end
+
+    case Bundles.list_bundles(flop_params, preload: [:uploaded_by_account, project: :account]) do
+      {bundles, meta} ->
+        conn
+        |> put_status(:ok)
+        |> json(%{
+          data: Enum.map(bundles, &bundle_list_to_map/1),
+          meta: %{
+            current_page: meta.current_page,
+            page_size: meta.page_size,
+            total_count: meta.total_count,
+            total_pages: meta.total_pages
+          }
+        })
+
+      error ->
+        conn
+        |> put_status(:internal_server_error)
+        |> json(%{error: "Failed to list bundles: #{inspect(error)}"})
+    end
+  end
+
+  operation :show,
+    summary: "Get a single bundle by ID",
+    operation_id: "getBundle",
+    parameters: %{
+      account_handle: [
+        in: :path,
+        type: :string,
+        required: true,
+        description: "The handle of the account."
+      ],
+      project_handle: [
+        in: :path,
+        type: :string,
+        required: true,
+        description: "The handle of the project."
+      ],
+      bundle_id: [
+        in: :path,
+        type: :string,
+        required: true,
+        description: "The ID of the bundle."
+      ]
+    },
+    responses: %{
+      ok: {"Bundle details", "application/json", TuistWeb.API.Schemas.Bundle},
+      not_found: {"Bundle not found", "application/json", Error},
+      unauthorized:
+        {"You need to be authenticated to view this bundle", "application/json", Error},
+      forbidden: {"You are not authorized to view this bundle", "application/json", Error}
+    }
+
+  def show(%{assigns: %{selected_project: selected_project}} = conn, %{"bundle_id" => bundle_id}) do
+    case Bundles.get_bundle(bundle_id, preload: [:uploaded_by_account, project: :account]) do
```

Comment:
```suggestion
    Bundles.get_bundle(bundle_id, project: selected_project, preload: [:uploaded_by_account, project: :account])
```

Instead of checking for the `project_id`, I'd search for a bundle with a given ID and a given project – if such a bundle is not found, we should return `:not_found` instead of `:forbidden`.